handlers/lobbies.js

```diff
@@ -22,19 +30,160 @@ Dota2._lobbyOptions = {
+ * @param {number} [options.bot_dire]
+ *
+ * Presumably the ID of the custom AI to be applied to dire bots.
+ *
+ * @param {boolean} [options.allow_spectating=true]
+ * @param {SeriesType} [options.series_type=NONE]
+ *
+ * Whether or not the game is part of a series (Bo3, Bo5).
+ *
+ * @param {number} [options.radiant_series_wins=0]
+ *
+ * # of games won so far, e.g. for a Bo3 or Bo5.
+ *
+ * @param {number} [options.dire_series_wins=0]
+ *
+ * # of games won so far, e.g. for a Bo3 or Bo5.
+ *
+ * @param {number} [options.previous_match_override]
+ *
+ * In a series, the match ID of the previous game. If not supplied, the GC will try
+ * to find it automatically based on the teams and the players.
+ *
+ * @param {boolean} [options.allchat=false]
+ * @param {LobbyDotaTVDelay} [options.dota_tv_delay=LobbyDotaTV_120]
+ *
+ * How much time the game should be delayed for DotaTV.
+ *
+ * @param {number} [options.leagueid]
+ *
+ * The league this lobby is being created for. The bot should be a league admin for
+ * this to work.
+ *
+ * @param {number} [options.custom_game_mode]
+ * @param {number} [options.custom_map_name]
+ * @param {number} [options.custom_difficulty]
+ * @param {number} [options.custom_game_id]
+ *
+ * @param {createPracticeLobbyCallback} [callback]
+ *
+ * The callback that handles the response.
+ */
+Dota2.Dota2Client.prototype.createPracticeLobby = function(options, callback) {
+    /**
+     * @callback createPracticeLobbyCallback
+     * @param {number|null} err
+     *
+     * Null if everything was successful, or a number corresponding to the error.
+     *
+     * @param {CMsgPracticeLobbyJoinResponse} response
+     *
+     * @typedef {object} CMsgPracticeLobbyJoinResponse
+     * @property {*} result
+     */
+    callback = callback || null;
+    var _self = this;
+
+    var defaults = {
+        game_name: "",
+        server_region: 0,
+        game_mode: Dota2.schema.lookupEnum("DOTA_GameMode").DOTA_GAMEMODE_AP,
+        game_version: Dota2.schema.lookupEnum("DOTAGameVersion").GAME_VERSION_STABLE,
+        cm_pick: Dota2.schema.lookupEnum("DOTA_CM_PICK").DOTA_CM_RANDOM,
+        allow_cheats: false,
+        fill_with_bots: false,
+        bot_difficulty_radiant: Dota2.BotDifficulty.PASSIVE,
+        bot_difficulty_dire: Dota2.BotDifficulty.PASSIVE,
+        allow_spectating: true,
+        pass_key: "",
+        series_type: Dota2.SeriesType.NONE,
+        radiant_series_wins: 0,
+        dire_series_wins: 0,
+        allchat: false,
+        dota_tv_delay: Dota2.schema.lookupEnum("LobbyDotaTVDelay").LobbyDotaTV_120,
+        leagueid: 0,
+        previous_match_override: 0,
+        custom_game_mode: 0,
+        custom_map_name: 0,
+        custom_difficulty: 0,
+        custom_game_id: 0
+    };
+    var finalOptions = Object.assign(defaults, options);
+
+    if (this.debug) util.log("Sending match CMsgPracticeLobbyCreate request");
+    var lobby_details = Dota2._parseOptions(finalOptions, Dota2._lobbyOptions);
```

Comment:
You need to make sure that `Dota2._lobbyOptions` contains a definition for all possible parameters, else this call will filter them out